internal/link/link.go

```diff
@@ -25,6 +26,7 @@ type Interface interface {
 	ForObjectWithQuery(object runtime.Object, text string, query url.Values) (*component.Link, error)
 	ForGVK(namespace, apiVersion, kind, name, text string) (*component.Link, error)
 	ForOwner(parent runtime.Object, controllerRef *metav1.OwnerReference) (*component.Link, error)
+	ForScaleTarget(parent runtime.Object, scaleTarget *v2beta2.CrossVersionObjectReference) (*component.Link, error)
```

Comment:
This is not general purpose enough to include in this interface. Link generation for scale targets should happen elsewhere.